aip/general/0123.md

```diff
@@ -71,6 +71,9 @@ message Topic {
 - Patterns **must** correspond to the [resource name][aip-122].
 - Pattern variables (the segments within braces) **must** use `snake_case`, and
   **must not** use an `_id` suffix.
+- Pattern variables **must** be the singular form of the resource type e.g. the
+  pattern variable representing a `Topic` resource ID is named `{topic}`.
+- Pattern variables **must** conform to the format `[_a-z]+`.
```

Comment:
This could probably be made tighter: `[a-z][_a-z]+[a-z]`

I don't think we'd want it to start or end with an underscore. The above prohibit single character pattern variables, admittedly. Do we definitely want to prohibit digits? I'm just wondering whether there might ever be something like `pop3_server` in there... hopefully not. (We can revisit, but then of course we'd need to check that there weren't implementations relying on there being no digits...)

Reply:
I guess collection identifiers are allowed to have digits (https://google.aip.dev/122#collection-identifiers), so that would imply that the resource type itself could have digits. I will update your expression to include digits in the middle and ending segements.

Reply:
I also changed the middle section to use `*` instead of `+` since that expression required it be 3 characters long. PTAL